File: Documentation/5.2/Samples/nodejs/ClientApi/Session/HowTo/getCurrentSessionNode.js
import * as assert from "assert";
import { DocumentStore } from "ravendb";

const store = new DocumentStore();
const session = store.openSession();

{
    //region current_session_node_1
    session.advanced.getCurrentSessionNode();
    //endregion
}

{
    /*
    //region current_session_node_2
    ServerNode {
        url: "https://localhost:8080",
        database: "Database",
        clusterTag: "A",
        serverRole: "Member" // Role can be one of: "None", "Promotable", "Member", "Rehab" 
    }
    //endregion
    */
}

async function examples() {
    {
        //region current_session_node_3
        const serverNode = session.advanced.getCurrentSessionNode();
        console.log(serverNode.url);
        //endregion
    }
}

File: Documentation/5.4/Samples/java/src/test/java/net/ravendb/ClientApi/Session/HowTo/GetCurrentSessionNode.java
package net.ravendb.ClientApi.Session.HowTo;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.client.http.ServerNode;

public class GetCurrentSessionNode {

    private interface IFoo {
        //region current_session_node_1
        ServerNode getCurrentSessionNode();
        //endregion
    }

    private static class Foo {
        //region current_session_node_2
        public class ServerNode {
            private String url;
            private String database;
            private String clusterTag;
            private Role serverRole;

            // getters and setters
        }

        public enum Role {
            NONE,
            PROMOTABLE,
            MEMBER,
            REHAB
        }
        //endregion
    }

    public void examples() {
        try (IDocumentStore store = new DocumentStore()) {
            try (IDocumentSession session = store.openSession()) {
                //region current_session_node_3
                ServerNode serverNode = session.advanced().getCurrentSessionNode();
                System.out.println(serverNode.getUrl());
                //endregion
            }
        }
    }
}
